File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    int count = 0;
    HashMap<Integer, Integer> hm;
    public int climbStairs(int n) {
        hm = new HashMap<>();
        dfs(0, n);
        return count;
    }
    
    public void dfs(int i, int n)
    {
        if (i == n)
        {
            count++;
            return;
        }
        if (i > n)
        {
            return;
        }
        if (hm.containsKey(i))
        {
            count += hm.get(i);
            return;
        }
        dfs(i+1, n);
        
        dfs(i+2, n);
        hm.put(i, count);
        return;
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        lists = [0] * (n+1)
        lists[n] = 1
        lists[n-1] = 1
        for i in range(n-2, -1, -1):
            lists[i] = lists[i+1] + lists[i+2]
        return lists[0]